src/EditorFeatures/CSharpTest/EditAndContinue/RudeEditTopLevelTests.cs

```diff
@@ -6709,6 +6709,64 @@ public void TupleNames_TypeUpdate()
         }
 
         [Fact]
+        public void TupleElementUpdate()
+        {
+            var src1 = "class C { (int, int) M() { return (1, 2); } }";
+            var src2 = "class C { (int, int) M() { return (1, 7); } }";
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifyEdits(
+                "Update [(int, int) M() { return (1, 2); }]@10 -> [(int, int) M() { return (1, 7); }]@10");
+
+            edits.VerifyRudeDiagnostics();
+        }
+
+        [Fact]
+        public void TupleElementDelete()
+        {
+            var src1 = "class C { (int, int, int a) M() { return (1, 2, 3); } }";
+            var src2 = "class C { (int, int) M() { return (1, 2); } }";
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifyEdits(
+                "Update [(int, int, int a) M() { return (1, 2, 3); }]@10 -> [(int, int) M() { return (1, 2); }]@10");
+
+            edits.VerifyRudeDiagnostics(
+                Diagnostic(RudeEditKind.TypeUpdate, "(int, int) M()", "method"));
+        }
+
+        [Fact]
+        public void TupleElementAdd()
+        {
+            var src1 = "class C { (int, int) M() { return (1, 2); } }";
+            var src2 = "class C { (int, int, int a) M() { return (1, 2, 3); } }";
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifyEdits(
+                "Update [(int, int) M() { return (1, 2); }]@10 -> [(int, int, int a) M() { return (1, 2, 3); }]@10");
+
+            edits.VerifyRudeDiagnostics(
+                Diagnostic(RudeEditKind.TypeUpdate, "(int, int, int a) M()", "method"));
+        }
+
+        [Fact]
+        public void ParenthesizedVariables_NamesUpdate()
+        {
+            var src1 = "class C { (int, int) M() { throw new System.Exception(); } void N() { int(a, b) = M(); } }";
+            var src2 = "class C { (int, int) M() { throw new System.Exception(); } void N() { int(b, c) = M(); } }";
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifyEdits(
+                "Update [void N() { int(a, b) = M(); }]@59 -> [void N() { int(b, c) = M(); }]@59");
+
+            edits.VerifyRudeDiagnostics();
+        }
```

Comment:
Thank you! Added

---
In reply to: [111841578](https://github.com/dotnet/roslyn/pull/18723#discussion_r111841578) [](ancestors = 111841578,111841504)